src/npc_attack.cpp

```diff
@@ -177,35 +211,62 @@ npc_attack_rating npc_attack_melee::evaluate_critter( const npc &source,
     const Creature::Attitude att = source.attitude_to( *critter );
     double potential = damage * npc_attack_constants::attitude_multiplier.at( att ) -
                        ( distance_to_me - 1 );
+
+    const int reach_range{ weapon.reach_range( source ) };
+    const int distance = rl_dist( source.pos(), critter->pos() );
+    int range_penalty = std::max( 0, distance - reach_range );
+    potential *= range_penalty ? 0.7f : 1.5f;
+
     if( damage >= critter->get_hp() ) {
         potential *= npc_attack_constants::kill_modifier;
     }
     if( target && target->pos() == critter->pos() ) {
         potential *= npc_attack_constants::target_modifier;
     }
+
     return npc_attack_rating( std::round( potential ), critter->pos() );
 }
 
 void npc_attack_gun::use( npc &source, const tripoint &location ) const
 {
-    const item &weapon = *gunmode;
-    if( !weapon.ammo_sufficient() ) {
-        source.do_reload( weapon );
+    if( !source.is_wielding( gun ) ) {
+        if( !source.wield( gun ) ) {
+            debugmsg( "ERROR: npc tried to equip a weapon it couldn't wield" );
+        }
+        return;
+    }
+
+    if( !gun.ammo_sufficient() ) {
+        source.do_reload( gun );
         add_msg_debug( debugmode::debug_filter::DF_NPC, "%s is reloading %s", source.disp_name(),
-                       weapon.display_name() );
+                       gun.display_name() );
+        return;
+    }
+
+    if( has_obstruction( source.pos(), location, false ) ||
+        ( source.rules.has_flag( ally_rule::avoid_friendly_fire ) &&
+          !source.wont_hit_friend( location, gun, false ) ) ) {
+        if( can_move( source ) ) {
+            source.avoid_friendly_fire();
+        } else {
+            source.move_pause();
+        }
         return;
     }
 
     const int dist = rl_dist( source.pos(), location );
 
-    if( source.aim_per_move( weapon, source.recoil ) > 0 &&
-        source.confident_shoot_range( weapon, source.get_most_accurate_sight( weapon ) ) >= dist ) {
+    // Only aim if we aren't in risk of being hit
+    // TODO: Get distance to closest enemy
```

Comment:
these comments refer to something that needs to be in evaluate(), not use().

Reply:
Evaluate won't really work for this though. Evaluate will give a target and preferred attack, but in some cases the NPC will choose to try and finish an enemy that isn't the closest and the NPC might get stuck trying to aim while being whacked by a nearby enemy constantly messing up the aim.

It has to be done here so the NPC tries to hipfire.